cockatrice/src/window_main.cpp

```diff
@@ -281,8 +281,17 @@ void MainWindow::actExit()
 
 void MainWindow::actAbout()
 {
+#if   defined(Q_PROCESSOR_X86_32)
```

Comment:
Shouldn't this section be invoking `getBuildArchitecture` instead of repeating the ifdef logic?

Reply:
In the same way that `getClientVersion` writes the string "Client Version: 2.4.1-custom_accd39f (2017-12-17)", `getBuildArchitecture` writes all three of the other lines. I think that having those #defines there is analogous to how the client version is added to the About dialog via `QString(" %1").arg(VERSION_STRING)`.

In the interest of having one set of `#ifdef` in one place for these, perhaps it makes more sense to add two more defined strings for `CLIENT_OS` and `BUILD_ARCHITECTURE` so that both `getBuildArchitecture` and `actAbout` are just returning whatever string is stored in those?

Reply:
Yes please, I really prefer to avoid propagating ifdefs around the codebase because it usually means we are missing some encapsulation or abstraction that would make things more readable and maintainable